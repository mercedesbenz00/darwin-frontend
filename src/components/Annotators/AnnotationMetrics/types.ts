import { MembershipPayload, DatasetAnnotationReportPayload } from '@/store/types'

export type AnnotationDataRange = 'total' | 'month' | 'week' | 'day'
export type AnnotationDataGranularity = 'day' | 'hour'

export type ChartDimension =
  'annotationTime' |
  'imagesAnnotated' |
  'avgTimePerAnnotation' |
  'annotationsCreated' |
  'reviewPassRate'

export type AdaptedDataPoint =
  {[k in Exclude<ChartDimension, 'reviewPassRate' | 'avgTimePerAnnotation'>]: number } &
  {[k in Extract<ChartDimension, 'reviewPassRate' | 'avgTimePerAnnotation'>]: number | null} &
  { userId: number | null, date: string }

export type AggregatedDataPoint = Pick<
    DatasetAnnotationReportPayload,
    | 'annotation_time'
    | 'annotations_approved'
    | 'annotations_created'
    | 'date'
    | 'images_annotated'
    | 'images_approved'
    | 'images_rejected'
    | 'user_id'
  >

export type AnnotationDimensionData = {
  annotationTime: number,
  annotationsCreated: number,
  avgTimePerAnnotation: number | null,
  imagesAnnotated: number,
  reviewPassRate: number | null
}

export type AnnotationChartData = {
  data: AdaptedDataPoint[]
  dimensions: AnnotationDimensionData
  member?: MembershipPayload
  visible: boolean
}

export type AnnotationBarChartData = {
  data: AggregatedDataPoint[]
  member?: MembershipPayload
}

/**
 * Format for a single item to be shown in a chart tooltip.
 *
 * A tooltip will contain multiple items
 */
export type ChartTooltipData = { member: MembershipPayload | null, value: string}
