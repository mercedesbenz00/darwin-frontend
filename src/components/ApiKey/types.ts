export type AbilityGroup =
  'manage_comments' |
  'manage_tasks' |
  'manage_datasets' |
  'manage_team' |
  'manage_team_members' |
  'manage_models'

export type Group = { id: AbilityGroup; name: string }
export type GroupOption = Group & { selected: boolean }

export enum SupportedLanguages {
  shell = 'shell',
  python = 'python',
  javascript = 'javascript',
  elixir = 'elixir',
  cli = 'cli'
}

export type AccessCodeExample = { language: SupportedLanguages, code: string }
