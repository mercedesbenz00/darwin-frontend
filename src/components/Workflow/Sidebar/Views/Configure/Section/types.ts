export type SidebarConfigureSectionTypes = {
  backgroundColor: string
  label: string
}
