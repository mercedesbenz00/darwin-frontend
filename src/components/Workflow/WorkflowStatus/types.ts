export enum WorkflowStatusVariant {
  ACTIVE = 'active',
  RUNNING = 'running',
  DRAFT = 'draft',
  OFFLINE = 'offline',
  INACTIVE = 'inactive'
}

export interface WorkflowStatusProps extends HTMLButtonElement {
  variant?: WorkflowStatusVariant
  dense?: boolean
}
