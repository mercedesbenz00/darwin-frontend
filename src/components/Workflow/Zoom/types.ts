export type ZoomProps = {
  playground: HTMLElement | null
  playgroundFrame: HTMLElement | null
  playgroundContent: HTMLElement | null
}
