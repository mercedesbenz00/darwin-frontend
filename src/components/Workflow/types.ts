export type Progress = {
  total: number,
  complete: number,
  // eslint-disable-next-line camelcase
  in_progress: number
  idle?: number
}
