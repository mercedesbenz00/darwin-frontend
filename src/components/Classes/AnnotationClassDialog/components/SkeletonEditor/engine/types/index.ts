export * from './SkeletonEdgeType'
export * from './SkeletonNodeType'
