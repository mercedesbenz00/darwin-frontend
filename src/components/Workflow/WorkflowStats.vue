<template>
  <div class="workflow-card__stats">
    <div class="workflow-card__stats__row workflow-card__stats__row__first">
      <attribute-stack
        class="workflow-card__stats__attribute-stack workflow-card__stats__attribute-stack__items"
      >
        <template #primary-attribute>
          {{ progress.total || '-' }}
        </template>
        <template #secondary-attribute>
          Items
        </template>
      </attribute-stack>
      <attribute-stack
        class="workflow-card__stats__attribute-stack
               workflow-card__stats__attribute-stack__complete"
      >
        <template #primary-attribute>
          {{ progress.complete || '-' }}
        </template>
        <template #secondary-attribute>
          Completed
        </template>
      </attribute-stack>
      <attribute-stack
        class="workflow-card__stats__attribute-stack
               workflow-card__stats__attribute-stack__in-progress"
      >
        <template #primary-attribute>
          {{ progress.inProgress || '-' }}
        </template>
        <template #secondary-attribute>
          In progress
        </template>
      </attribute-stack>
    </div>
    <div class="workflow-card__stats__row workflow-card__stats__row__second">
      <div
        v-tooltip="{
          content: progressTooltip,
          placement: 'top',
          delay: { show: 300 },
          classes: 'tooltip--workflow-card'
        }"
        class="workflow-card__stats__progress"
      >
        <div class="workflow-card__stats__progress__bar">
          <progress-bar :value="progressValue" />
        </div>
        <div class="workflow-card__stats__progress__label">
          {{ progressPercentage }}%
        </div>
      </div>
    </div>
  </div>
</template>

<script lang="ts">
import { Component, Prop, Vue } from 'vue-property-decorator'

import AttributeStack from '@/components/Common/AttributeStack/AttributeStack.vue'
import ProgressBar from '@/components/Common/ProgressBar/ProgressBar.vue'
import { formatNumericValue } from '@/utils'

import { Progress } from './types'

@Component({
  name: 'workflow-stats',
  components: { AttributeStack, ProgressBar }
})
export default class WorkflowCard extends Vue {
  @Prop({ type: Object })
  value!: Progress

  get progress (): any {
    const { total, complete, in_progress: inProgress } = this.value ||
      { total: 0, complete: 0, in_progress: 0 }
    return {
      total: formatNumericValue(total),
      complete: formatNumericValue(complete),
      inProgress: formatNumericValue(inProgress)
    }
  }

  get progressValue (): number {
    const { total, complete } = this.value ||
      { total: 0, complete: 0 }
    if (!total) { return 0 }
    return (complete / total)
  }

  get progressPercentage (): string {
    return (this.progressValue * 100).toFixed(1)
  }

  get progressTooltip (): string {
    if (!this.progressPercentage) { return '' }
    return `${this.progressPercentage}% Completed`
  }
}
</script>

<style lang="scss" scoped>
.workflow-card__stats {
  @include col;
  align-items: stretch;
  gap: 2px;

  &__row {
    @include row--distributed;

    &__first {
      gap: 2px;
      margin-bottom: 4px;
    }
  }

  &__attribute-stack {
    height: 100%;
    width: 100%;
  }

  &__progress {
    @include row--distributed;
    align-items: center;
    width: 100%;

    &__bar {
      width: 100%;
    }

    &__label {
      margin-left: 12px;
      text-transform: capitalize;
      color: $colorContentTertiary;
      @include typography(md, inter);
    }
  }
}
</style>
