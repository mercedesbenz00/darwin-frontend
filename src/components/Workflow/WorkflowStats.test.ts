import { createLocalVue, shallowMount, mount, Wrapper } from '@vue/test-utils'

import { stubDirectiveWithAttribute } from 'test/unit/directiveStubs'

import WorkflowStats from '@/components/Workflow//WorkflowStats.vue'

import { Progress } from './types'

const localVue = createLocalVue()
localVue.directive('tooltip', stubDirectiveWithAttribute)

type WorkflowStatsProp = {
  value: Progress,
}

let propsData: WorkflowStatsProp

const itMatchesSnapshot = (propsData: WorkflowStatsProp): void => it('matches snapshot', () => {
  const wrapper = mount(WorkflowStats, { localVue, propsData })
  expect(wrapper).toMatchSnapshot()
})

const progress = {
  total: 10000,
  complete: 1000,
  in_progress: 9000,
  idle: 0
}

itMatchesSnapshot({
  value: progress
})

it('is instanciated', () => {
  const _propsData = { value: progress }

  const wrapper = shallowMount(WorkflowStats, { localVue, propsData: _propsData })
  expect(wrapper).toBeTruthy()
})

describe('when correct props are used', () => {
  let wrapper: Wrapper<Vue>

  const _propsData = { value: progress }

  beforeEach(() => {
    propsData = _propsData
    wrapper = mount(WorkflowStats, { localVue, propsData })
  })

  afterEach(() => {
    wrapper.destroy()
  })

  itMatchesSnapshot(_propsData)

  it('should show correct progress items', () => {
    const attributeStack = wrapper
      .find('.workflow-card__stats__attribute-stack__items')
      .find('.attribute-stack')
    expect(attributeStack.find('.attribute-stack__label--primary').text()).toBe('10.0k')
  })

  it('should show correct progress complete', () => {
    const attributeStack = wrapper
      .find('.workflow-card__stats__attribute-stack__complete')
      .find('.attribute-stack')
    expect(attributeStack.find('.attribute-stack__label--primary').text()).toBe('1000')
  })

  it('should show correct progress complete', () => {
    const attributeStack = wrapper
      .find('.workflow-card__stats__attribute-stack__in-progress')
      .find('.attribute-stack')
    expect(attributeStack.find('.attribute-stack__label--primary').text()).toBe('9000')
  })

  it('should show correct progress percentage', () => {
    const progress = wrapper.find('.workflow-card__stats__progress')
    expect(progress.find('.workflow-card__stats__progress__label').text()).toBe('10.0%')
  })
})
