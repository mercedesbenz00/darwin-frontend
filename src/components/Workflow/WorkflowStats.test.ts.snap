// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div class="workflow-card__stats">
  <div class="workflow-card__stats__row workflow-card__stats__row__first">
    <div class="attribute-stack workflow-card__stats__attribute-stack workflow-card__stats__attribute-stack__items">
      <h3 class="attribute-stack__label attribute-stack__label--primary">
        10.0k
      </h3>
      <h3 class="attribute-stack__label attribute-stack__label--secondary">
        Items
      </h3>
    </div>
    <div class="attribute-stack workflow-card__stats__attribute-stack workflow-card__stats__attribute-stack__complete">
      <h3 class="attribute-stack__label attribute-stack__label--primary">
        1000
      </h3>
      <h3 class="attribute-stack__label attribute-stack__label--secondary">
        Completed
      </h3>
    </div>
    <div class="attribute-stack workflow-card__stats__attribute-stack workflow-card__stats__attribute-stack__in-progress">
      <h3 class="attribute-stack__label attribute-stack__label--primary">
        9000
      </h3>
      <h3 class="attribute-stack__label attribute-stack__label--secondary">
        In progress
      </h3>
    </div>
  </div>
  <div class="workflow-card__stats__row workflow-card__stats__row__second">
    <div class="workflow-card__stats__progress" tooltip="{'content':'10.0% Completed','placement':'top','delay':{'show':300},'classes':'tooltip--workflow-card'}">
      <div class="workflow-card__stats__progress__bar">
        <div role="progressbar" tabindex="1" class="progress-indicator"><span class="progress-indicator-value progress-indicator-value--active" style="width: 10%;"></span></div>
      </div>
      <div class="workflow-card__stats__progress__label">
        10.0%
      </div>
    </div>
  </div>
</div>
`;

exports[`when correct props are used matches snapshot 1`] = `
<div class="workflow-card__stats">
  <div class="workflow-card__stats__row workflow-card__stats__row__first">
    <div class="attribute-stack workflow-card__stats__attribute-stack workflow-card__stats__attribute-stack__items">
      <h3 class="attribute-stack__label attribute-stack__label--primary">
        10.0k
      </h3>
      <h3 class="attribute-stack__label attribute-stack__label--secondary">
        Items
      </h3>
    </div>
    <div class="attribute-stack workflow-card__stats__attribute-stack workflow-card__stats__attribute-stack__complete">
      <h3 class="attribute-stack__label attribute-stack__label--primary">
        1000
      </h3>
      <h3 class="attribute-stack__label attribute-stack__label--secondary">
        Completed
      </h3>
    </div>
    <div class="attribute-stack workflow-card__stats__attribute-stack workflow-card__stats__attribute-stack__in-progress">
      <h3 class="attribute-stack__label attribute-stack__label--primary">
        9000
      </h3>
      <h3 class="attribute-stack__label attribute-stack__label--secondary">
        In progress
      </h3>
    </div>
  </div>
  <div class="workflow-card__stats__row workflow-card__stats__row__second">
    <div class="workflow-card__stats__progress" tooltip="{'content':'10.0% Completed','placement':'top','delay':{'show':300},'classes':'tooltip--workflow-card'}">
      <div class="workflow-card__stats__progress__bar">
        <div role="progressbar" tabindex="1" class="progress-indicator"><span class="progress-indicator-value progress-indicator-value--active" style="width: 10%;"></span></div>
      </div>
      <div class="workflow-card__stats__progress__label">
        10.0%
      </div>
    </div>
  </div>
</div>
`;
